### PR TITLE
Show cache key when logging compilation cache hits/misses

### LINKED ISSUE #22548: When logging cache hit/miss, include cache_key
I want these log messages:

```
Persistent compilation cache hit for 'jit_scan_outer_bigs'
```

To look like this:
```
Persistent compilation cache hit for 'jit_scan_outer_bigs' with cache_key `b86d2a45745d2417b7a28b1c38aa39497bf611e8320238cabe31e99142be5f2d`
```

Optionally we can truncate the cache key. I implemented this feature, let me know if you want a pull request.

### PR DESCRIPTION
**Pull request by cool-RR**
Fixes #22548

### CODE REVIEW

**Comment by cool-RR** [comment]
Example log lines (not in chronological order):

```
2024-07-22 11:56:03,008 - DEBUG - Writing jit_matrix_multiply to persistent compilation cache with key 'jit_matrix_multiply-fa478e4ead6882a4c14cc1366d30f8efc783165c9cb16536e025357428373990'
2024-07-22 12:24:05,601 - DEBUG - PERSISTENT COMPILATION CACHE MISS for 'jit_convert_element_type' with key 'jit_convert_element_type-92121ef34fa61ce74b4dad69c9ff3d93d790dbdf2ac0c7664ab8933c8cb46c27'
2024-07-22 12:24:05,834 - DEBUG - PERSISTENT COMPILATION CACHE MISS for 'jit_matrix_multiply' with key 'jit_matrix_multiply-fa478e4ead6882a4c14cc1366d30f8efc783165c9cb16536e025357428373990'
2024-07-22 12:23:09,163 - DEBUG - Persistent compilation cache hit for 'jit_matrix_multiply' with key 'jit_matrix_multiply-fa478e4ead6882a4c14cc1366d30f8efc783165c9cb16536e025357428373990'
```

**Comment by cool-RR** [comment]
@ayaka14732 Could you enable the tests to run here please?

File: jax/_src/compiler.py [hunk 1]
```diff
@@ -91,21 +91,21 @@ def use_detailed_logging(module: ir.Module) -> bool:
...
 
 
-def log_persistent_cache_hit(module_name: str) -> None:
+def log_persistent_cache_hit(module_name: str, cache_key: str) -> None:
   hit_log_priority = (logging.WARNING if config.log_compiles.value
                       else logging.DEBUG)
-  logger.log(hit_log_priority, "Persistent compilation cache hit for '%s'",
-             module_name)
+  logger.log(hit_log_priority, "Persistent compilation cache hit for '%s' with key %r",
+             module_name, cache_key)
```
Comment (yashk2810) [inline jax/_src/compiler.py:98]: I don't think we should print the cache_key here. Above is fine, but here we should explain why there was a cache miss.

Comment (cool-RR) [reply jax/_src/compiler.py:98]: I do think it's useful to know which cache file was used when debugging caching.

I don't understand the comment "here we should explain why there was a cache miss" given that we're talking about a cache hit.

Comment (yashk2810) [reply jax/_src/compiler.py:98]: But seems like you are still printing the cache key for `miss` too. Is that enough information for you to figure out what led to the cache miss?

Comment (cool-RR) [reply jax/_src/compiler.py:98]: What I can do with this information is look at the folder and check whether a file with that key exists. If it does, then maybe the folder wasn't configured correctly.

While we could add more useful information, it's beyond the scope of what I'm interested in implementing in this PR.

Comment (yashk2810) [reply jax/_src/compiler.py:98]: Okay, sounds reasonable.

**Review by yashk2810 — APPROVED** [review]

**Comment by cool-RR** [comment]
I believe the CI failure is not related to my changes, as I'm seeing similar failures in other PRs.

**Review by ayaka14732 — APPROVED** [review]

**Pull request merged**